frontend/src/components/module-viewer/content/modals/EditTextModal.tsx

```diff
@@ -1,25 +1,169 @@
+import React, { useState, useCallback, useMemo } from "react";
+import isHotkey, { isKeyHotkey } from "is-hotkey";
+import { createEditor, Transforms, Range } from "slate";
+import { Slate, Editable, withReact } from "slate-react";
+import { withHistory } from "slate-history";
+import isUrl from "is-url";
+import {
+  RiAlignLeft,
+  RiAlignCenter,
+  RiAlignRight,
+  RiAlignJustify,
+  RiBold,
+  RiItalic,
+  RiUnderline,
+} from "react-icons/ri";
+
 import { TextBlockState } from "../../../../types/ContentBlockTypes";
 import { Modal } from "../../../common/Modal";
+import FormatButton, { toggleFormat } from "./FormatButton";
+import AlignmentButton from "./AlignmentButton";
+import LinkButton, { wrapLink } from "./LinkButton";
+
+import {
+  EditContentModalProps,
+  CustomElement,
+  TextEditor,
+  LeafPropTypes,
+  ElementPropTypes,
+  FormatEnum,
+} from "../../../../types/ModuleEditorTypes";
+
+const HOTKEYS: Record<string, FormatEnum> = {
+  "mod+b": "bold",
+  "mod+i": "italic",
+  "mod+u": "underline",
+};
+
+const Leaf = ({ attributes, children, leaf }: LeafPropTypes) => {
+  return (
+    <span
+      {...attributes}
+      style={{
+        fontWeight: leaf.bold ? "bold" : "normal",
+        fontStyle: leaf.italic ? "italic" : "normal",
+        textDecorationLine: leaf.underline ? "underline" : "none",
+      }}
+    >
+      {children}
+    </span>
+  );
+};
+
+const Element = ({ attributes, children, element }: ElementPropTypes) => {
+  switch (element.type) {
+    case "link":
+      return (
+        <Link
+          {...attributes}
+          style={{ textDecorationLine: "underline" }}
+          color="purple"
+          isExternal
+          href={element.url}
+          onClick={() => {
+            window.open(element.url, "_blank", "noopener,noreferrer");
+          }}
+        >
+          {children}
+        </Link>
+      );
+    default:
+      return (
+        <p style={{ textAlign: element.align }} {...attributes}>
+          {children}
+        </p>
+      );
+  }
+};
+
+const initialValue: Array<CustomElement> = [
+  {
+    type: "paragraph",
+    align: "left",
+    children: [{ text: "" }],
+  },
+];
+
+/* eslint-disable no-param-reassign */
+const withInlines = (editor: TextEditor) => {
+  const { insertData, insertText, isInline } = editor;
+
+  editor.isInline = (element: CustomElement) =>
+    ["link"].includes(element.type) || isInline(element);
+
+  editor.insertText = (text) => {
+    if (text && isUrl(text)) {
+      wrapLink(editor, text);
+    } else {
+      insertText(text);
+    }
+  };
+
+  editor.insertData = (data) => {
+    const text = data.getData("text/plain");
+
+    if (text && isUrl(text)) {
+      wrapLink(editor, text);
+    } else {
+      insertData(data);
+    }
+  };
 
-import { TextArea } from "../../../common/TextArea";
-import { EditContentModalProps } from "../../../../types/ModuleEditorTypes";
+  return editor;
+};
+/* eslint-enable no-param-reassign */
```

Comment:
Is this line necessary? It's not clear to me what it's applying to but maybe I just missed it.

Reply:
it's applying to any line where we make an assignment to an editor property.
ex.
```
editor.isInline = ...
editor.insertData = ...
```
I didn't really understand why it was a problem, but I went with it - hoping to ask later